### PR TITLE
Send function app directory at specialization

### PR DESCRIPTION
**Pull request by mhoeger**
Resolves https://github.com/Azure/azure-functions-host/issues/5238

### CODE REVIEW

File: src/WebJobs.Script.Grpc/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -189,6 +189,8 @@ message WorkerStatusResponse {
 message FunctionEnvironmentReloadRequest {
   // Environment variables from the current process
   map<string, string> environment_variables = 1;
+  // Current directory of function app
+  string current_directory = 2;
```
Comment (pragnagopa) [inline src/WebJobs.Script.Grpc/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: using old proto file? shouln't this be function_app_directory?

Comment (mhoeger) [reply src/WebJobs.Script.Grpc/azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: oh shoot! thank you!

File: src/WebJobs.Script.WebHost/Standby/StandbyManager.cs [hunk 2]
```diff
@@ -107,13 +107,13 @@ public async Task SpecializeHostCoreAsync()
             // user dependencies
             FunctionAssemblyLoadContext.ResetSharedContext();
 
+            NotifyChange();
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Standby/StandbyManager.cs]: please add a comment why NotifyChange needs to be called before specializing language workers

File: test/WebJobs.Script.Tests/Workers/Rpc/RpcWorkerChannelTests.cs [hunk 3]
```diff
@@ -248,6 +260,19 @@ public void SendSendFunctionEnvironmentReloadRequest_SanitizedEnvironmentVariabl
...
+            var environmentVariables = new Dictionary<string, string>()
+            {
+                { "TestValid", "TestValue" }
+            };
+
+            FunctionEnvironmentReloadRequest envReloadRequest = _workerChannel.GetFunctionEnvironmentReloadRequest(environmentVariables);
+            Assert.True(envReloadRequest.EnvironmentVariables["TestValid"] == "TestValue");
+            Assert.True(envReloadRequest.CurrentDirectory == _scriptRootPath);
+        }
+
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests/Workers/Rpc/RpcWorkerChannelTests.cs]: Also, take a look at tests here: https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Host/StandbyManager/StandbyManagerE2ETests_Windows.cs Please add a test for node.

**Comment by mhoeger** [comment]
@pragnagopa - have verified on private stamp. can you look at this again?

**Comment by pragnagopa** [comment]
@mhoeger - StandbyManager test is failing. In the past, these tests were flaky and I have disabled Java E2E tests as well. Good to have this test in place. For now, please disable the test and update issue #4230

**Comment by mhoeger** [comment]
Done! @pragnagopa - could you approve again?

**Pull request merged**